### DISCUSSION TITLE
能否增加对国产矽昌SF19A28芯片的支持？

### DISCUSSION THREAD

**Discussion by xiamiwolf** [messense/aliyundrive-webdav#485] [Q&A] [answered]
GL.iNet SF1200无线路由器使用国产矽昌SF19A28芯片，虽然也是mips架构，但由于经过了魔改，不论是mips_24kc还是mipsel_24kc的elf均无法在其上运行。虽然小众了一点，但GL.iNet提供了完整的源代码，希望能够增加对国产芯片的支持，不甚感激！

**Comment by messense** [comment]
目前应该支持不了，需要 [LLVM](https://llvm.org/) 和 [Rust](https://www.rust-lang.org/) 先支持，不过这种魔改架构，除非官方去给 LLVM 和 Rust 增加支持，几乎没有人会做的。

**Reply by xiamiwolf** [reply]
据我在GL-iNet论坛所了解，有人已通过Github Action云编译为SFT1200成功编译了aliyundrive-webdav，且能正确使用。（详见此帖：[sft1200软件云编译](https://forum.gl-inet.cn/forum.php?mod=viewthread&tid=251&extra=page%3D2)），希望能够有所帮助。

**Reply by messense** [reply]
https://github.com/Misaka-Aki/openwrt-sdk-ACTION 404 not found.

**Reply by xiamiwolf** [reply]
不知道作者为什么把那个ACTION库删了，正想办法看能不能联系上。

**Reply by messense** [reply]
https://forum.gl-inet.cn/forum.php?mod=viewthread&tid=352&extra=page%3D2

有人编译了。

**Comment by messense** [comment, marked as answer]
试试这个。
[aliyundrive-webdav_1.8.4_mips_siflower.ipk.zip](https://github.com/messense/aliyundrive-webdav/files/9173345/aliyundrive-webdav_1.8.4_mips_siflower.ipk.zip)

**Reply by xiamiwolf** [reply]
非常感谢！刚利用GLiNet论坛提供的Action编译成功了！
再次万分感谢您将国产矽昌SF19A28芯片纳入支持，也希望我们的国产化能得到越来越多的人的支持。
